src/Common/tests/System/Collections/ICollection.NonGeneric.Tests.cs

```diff
@@ -0,0 +1,271 @@
+            Assert.NotEqual(collection1.SyncRoot, collection2.SyncRoot);
+        }
+
+        #endregion
+
+        #region CopyTo
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_NonGeneric_CopyTo_NullArray_ThrowsArgumentNullException(int count)
+        {
+            ICollection collection = NonGenericICollectionFactory(count);
+            Assert.Throws<ArgumentNullException>(() => collection.CopyTo(null, 0));
+        }
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_NonGeneric_CopyTo_TwoDimensionArray_ThrowsArgumentException(int count)
+        {
+            if (count > 0)
+            {
+                ICollection collection = NonGenericICollectionFactory(count);
+                Array arr = new object[count,count];
+                Assert.Equal(2, arr.Rank);
+                Assert.Throws<ArgumentException>(() => collection.CopyTo(arr, 0));
+            }
+        }
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_NonGeneric_CopyTo_NonZeroLowerBound(int count)
+        {
+            ICollection collection = NonGenericICollectionFactory(count);
+            Array arr = Array.CreateInstance(typeof(object), new int[1] { 2 }, new int[1] { 2 });
+            Assert.Equal(1, arr.Rank);
+            Assert.Equal(2, arr.GetLowerBound(0));
+            Assert.ThrowsAny<ArgumentException>(() => collection.CopyTo(arr, 0));
+        }
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public virtual void ICollection_NonGeneric_CopyTo_ArrayOfIncorrectValueType(int count)
+        {
+            if (count > 0)
+            {
+                ICollection collection = NonGenericICollectionFactory(count);
+                float[] array = new float[count * 3 / 2];
+                Assert.Throws<ArgumentException>(() => collection.CopyTo(array, 0));
+            }
+        }
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_NonGeneric_CopyTo_ArrayOfIncorrectReferenceType(int count)
+        {
+            if (count > 0)
+            {
+                ICollection collection = NonGenericICollectionFactory(count);
+                StringBuilder[] array = new StringBuilder[count * 3 / 2];
+                Assert.Throws<ArgumentException>(() => collection.CopyTo(array, 0));
+            }
+        }
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public virtual void ICollection_NonGeneric_CopyTo_ArrayOfEnumType(int count)
+        {
+            Array enumArr = Enum.GetValues(typeof(EnumerableType));
+            if (count > 0 && count < enumArr.Length)
+            {
+                ICollection collection = NonGenericICollectionFactory(count);
+                // Implementations may throw either an InvalidCastExcepton or an ArgumentException.
```

Comment:
Does this differ per collection type?  Is it a bug, or this is just expected?  Can we specify to the test which exception is expected so we can lock in that behavior and be alerted to it changing in the future?